pennylane/labs/resource_estimation/resource_operator.py

```diff
@@ -95,52 +95,97 @@ def set_resources(cls, new_func: Callable) -> None:
     @property
     @abstractmethod
     def resource_params(self) -> dict:
-        """Returns a dictionary containing the minimal information needed to
+        r"""Returns a dictionary containing the minimal information needed to
         compute a compressed representation"""
 
     @classmethod
     @abstractmethod
     def resource_rep(cls, *args, **kwargs) -> CompressedResourceOp:
-        """Returns a compressed representation containing only the parameters of
+        r"""Returns a compressed representation containing only the parameters of
         the Operator that are needed to compute a resource estimation."""
 
     def resource_rep_from_op(self) -> CompressedResourceOp:
-        """Returns a compressed representation directly from the operator"""
+        r"""Returns a compressed representation directly from the operator"""
         return self.__class__.resource_rep(**self.resource_params)
 
     @classmethod
     def adjoint_resource_decomp(cls, *args, **kwargs) -> Dict[CompressedResourceOp, int]:
-        """Returns a compressed representation of the adjoint of the operator"""
+        r"""Returns a dictionary representing the resources for the adjoint of the operator.
+
+        Raises:
+            ResourcesNotDefined: no resources implemented by default
+
+        Returns:
+            Dict[CompressedResourceOp, int]: The keys are the operators and the associated
+                values are the counts.
+        """
         raise ResourcesNotDefined
 
     @classmethod
     def controlled_resource_decomp(
-        cls, num_ctrl_wires, num_ctrl_values, num_work_wires, *args, **kwargs
+        cls, num_ctrl_wires: int, num_ctrl_values: int, num_work_wires: int, *args, **kwargs
     ) -> Dict[CompressedResourceOp, int]:
-        """Returns a compressed representation of the controlled version of the operator"""
+        r"""Returns a dictionary representing the resources for a controlled version of the operator.
+
+        Args:
+            num_ctrl_wires (int): the number of qubits the operation is controlled on
+            num_ctrl_values (int): the number of control qubits, that are controlled when off
+            num_work_wires (int): the number of additional qubits that can be used for decomposition
+
+        Raises:
+            ResourcesNotDefined: no resources implemented by default
+
+        Returns:
+            Dict[CompressedResourceOp, int]: The keys are the operators and the associated
+                values are the counts.
```

Comment:
Maybe should just say raises?

Reply:
Same with all the other functions.

Reply:
I think it's fine to leave the return for the base doc-string.